pkg/cloud/gcp/actuators/machine/reconciler_test.go

```diff
@@ -89,3 +87,24 @@ func TestReconcileMachineWithCloudState(t *testing.T) {
 		t.Errorf("Expected: %s, got: %s", instanceName, *r.providerStatus.InstanceID)
 	}
 }
+
+func TestExists(t *testing.T) {
+	_, mockComputeService := computeservice.NewComputeServiceMock()
+	machineScope := machineScope{
+		machine: &machinev1beta1.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "",
+				Namespace: "",
+			},
+		},
+		coreClient:     controllerfake.NewFakeClient(),
+		providerSpec:   &gcpv1beta1.GCPMachineProviderSpec{},
+		providerStatus: &gcpv1beta1.GCPMachineProviderStatus{},
+		computeService: mockComputeService,
+	}
+	reconciler := newReconciler(&machineScope)
+	exists, err := reconciler.instanceExists()
+	if err != nil || exists != true {
```

Comment:
We should separate these two conditions. 
- not expected to return an error
- not expected to be false

Reply:
@frobware this is in complete, MVP implementation to enable other work :)